controller/resource_roles.go

```diff
@@ -87,6 +91,74 @@ func (c *ResourceRolesController) ListAssignedByRoleName(ctx *app.ListAssignedBy
 	})
 }
 
+// AssignRole assigns a specific role for a resource, to one or more identities.
+func (c *ResourceRolesController) AssignRole(ctx *app.AssignRoleResourceRolesContext) error {
+
+	currentUser, err := login.ContextIdentity(ctx)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, errors.NewUnauthorizedError(err.Error()))
+	}
+	if currentUser == nil {
+		return jsonapi.JSONErrorResponse(ctx, errors.NewUnauthorizedError("identity ID not found in token"))
```

Comment:
Added a test https://github.com/fabric8-services/fabric8-auth/pull/481/files#diff-4bd1c185cd704de02aaa93a028975a89R316 for this but doesn't show up in coverage. Must be missing something very obvious.

Reply:
ContextIdentity() will return an error if there is no "sub" claim in the token.
So, if err=="nil" then currentUser != nil. You can just remove this check.